src/engines/kotorbase/module.cpp

```diff
@@ -673,6 +677,17 @@ void Module::handleActions() {
 	}
 }
 
+void Module::handleHeartbeat() {
+	const int kHeartbeatInterval = 1000; // ms
```

Comment:
Is every second correcct for KotOR? In NWN, the heartbeat is invoked every 6 seconds (= the length of a combat round)

Reply:
I'm not really sure. What would be the right place to look for a reference?

Reply:
On Deadly Stream, a KotOR modding site, they also say 6s: https://deadlystream.com/topic/6371-on-heartbeat/ (the Lexicon is https://nwnlexicon.com/) , https://deadlystream.com/topic/2062-tsl-textures-remastered/?page=9&tab=comments#comment-23797

Reply:
In the future, we should probably have a "round handler" or something in the KotOR engine, that synchronously does all the things required per round. I.e. calling all heartbeat scripts, evaluating pathfinding, checking and firing script events like perception, going through combat initiative, etc.

Reply:
Changed the interval to 6 seconds.